mne/io/eeglab/tests/test_eeglab.py

```diff
@@ -308,4 +309,35 @@ def test_eeglab_event_from_annot():
     assert len(events_b) == 154
 
 
+def _fake_montage(ch_names):
+    return Montage(
+        pos=np.random.RandomState(42).randn(len(ch_names), 3),
+        ch_names=ch_names,
+        # kind=4,
+        kind='foo',
+        selection=np.arange(len(ch_names))
+    )
+
+
+@testing.requires_testing_data
+def test_montage():
+    """Test montage."""
+    base_dir = op.join(testing.data_path(download=False), 'EEGLAB')
+    fname = op.join(base_dir, 'test_raw.set')
+    # montage = op.join(base_dir, 'test_chans.locs')
+
+    raw_none = read_raw_eeglab(input_fname=fname, montage=None, preload=False)
+    montage = _fake_montage(raw_none.info['ch_names'])
+
+    raw_montage = read_raw_eeglab(input_fname=fname, montage=montage,
+                                  preload=False)
+    raw_none.set_montage(montage)
+    assert object_diff(raw_none.info['dig'], raw_montage.info['dig']) == ''
+    # assert object_diff(raw_none.info['chs'], raw_montage.info['chs']) == ''
+    diff = object_diff(raw_none.info['chs'],
+                       raw_montage.info['chs']).splitlines()
+    for dd in diff:
+        assert 'coord_frame' in dd
```

Comment:
At this point, the `coord_frame` do not match, and IMHO they should. But this can be done in a separate PR to chase why.

At this point, we have refactored the `RawEEGLAB.__init__` to have `set_montage` at the end without breaking any test (and hopefully no users code)

Reply:
> At this point, we have refactored the RawEEGLAB.__init__ to have set_montage at the end without breaking any test (and hopefully no users code)

I don't think this is true actually, the code above that you said is in `EpochsEEGLAB` appears to be in `RawEEGLAB`. But I'm okay with merging this as is, and having some potentially breaking changes in a subsequent PR to get `montage` only used at the end of `__init__` for both EpochsEEGLAB and RawEEGLAB variants.